src/index.ts

```diff
@@ -76,4 +77,61 @@ app.post("/logout", async(c) => {
   return new Response();
 });
 
+app.post("/attendance", async(c) => {
+  /*ネットワークの確認を行う*/
+  const allowPattern = new RegExp(c.env.ALLOW_IP_REGEX);
+  if ( !allowPattern.test(c.req.header("cf-connectiong-ip") ?? "") ) { //"??":c.req.header(cf-connectiong-ip")でnullが返ってきたときに""として扱う演算子
+    return c.text("Forbidden", 403);
+  }
+
+  /*学生の認証を行う*/
+  //sessionがcookieあるか確認
+  const session = c.get("session");
+  if ( !session ){
+    return c.text("Unauthorized", 401);
+  }
+
+  /*まとめてクエリを実行する*/
+  const now = new Date();
+  const [ existResult, accountEntry, attendanceBoardEntry ] = await c.env.DB.batch([
+    c.env.DB.prepare("SELECT * FROM session WHERE id = ?").bind(session.id),
+    c.env.DB.prepare("SELECT * FROM account WHERE role = 'STUDENT' AND id = ?").bind(session.account_id),
+    c.env.DB.prepare("SELECT id FROM attendance_board WHERE start_from  <= ?1 AND seconds_from_be_late_to_end >= ?2").bind(now, now),
+  ]);
+  //session情報がsessionテーブルにある確認
+  const exists = existResult !== null;
+  if ( !exists ){
+    return c.text("Unauthorized", 401);
+  }
+  //roleが学生か確認
+  const isStudent = accountEntry !== null;
+  if (!isStudent) {
+    return c.text("Unauthorized", 401);
+  }
+
+  /*出席申請受付が開始されている科目をDBから探す、存在しなければ404を返す*/
+  const isSubjectNone = attendanceBoardEntry == null;
+  if ( isSubjectNone ){
+    return c.text("Not Found", 404);
+  }
+
+  /*まとめてクエリを実行する*/
+  const [ attendResult, request ] = await c.env.DB.batch([
+    c.env.DB.prepare("SELECT * FROM attendance WHERE \"where\" = ?").bind(attendanceBoardEntry),
+    c.env.DB.prepare("INSERT INTO attendance (id, create_at, who, \"where\") VALUES (?1, ?2, ?3, ?4)").bind(nanoid(), new Date(), session.account_id, attendanceBoardEntry)
```

Comment:
```suggestion
    c.env.DB.prepare("INSERT INTO attendance (id, create_at, who, \"where\") VALUES (?1, ?2, ?3, ?4)").bind(nanoid(), now, session.account_id, attendanceBoardEntry)
```
この上で取得している処理開始時点の現在時刻 `now` を使用しないと, データに矛盾が発生するおそれがあります.